src/nupic/algorithms/Connections.cpp

```diff
@@ -83,11 +83,13 @@ void Connections::unsubscribe(UInt32 token) {
 
 Segment Connections::createSegment(CellIdx cell) {
   Segment segment;
-  if (destroyedSegments_.size() > 0) {
+  if (!destroyedSegments_.empty() ) {
     segment = destroyedSegments_.back();
     destroyedSegments_.pop_back();
   } else {
-    segment = (Segment)segments_.size();
+    NTA_CHECK(segments_.size() <= std::numeric_limits<Segment>::max()) << "Add segment failed: Range of Segment (data-type) insufficinet size."
+	    << segments_.size() << " <= " << std::numeric_limits<Segment>::max();
+    segment = static_cast<Segment>(segments_.size());
     segments_.push_back(SegmentData());
```

Comment:
If before this call: segments_.size() == numeric_limits<Segment>::max()
Then after this call: segments_.size() > numeric_limits<Segment>::max()
Instead it should raise an exception.

Reply:
thanks, I see. I should not allow push_back new segment if size() already == limits_max.

Reply:
@ctrl-z-9000-times fixed the `<` and actually added tests that verify the overflow would be detected.